### PR TITLE
feat: promote ingress settings to public

### PR DESCRIPTION
**Pull request by jawnsy**
* Add ingress section to values.yaml and re-generate documentation
* Add tests for ingress settings
* Change comment syntax to {{/* */}} to prevent it from appearing in the generated Ingress spec

### CODE REVIEW

**Comment by jawnsy** [comment]
This has some of the changes in #169 (namely, exposing the ingress in our docs), but I split it into a separate PR to make it less risky and easier to review

File: tests/ingress_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+			AssertFunc: func(t *testing.T, ingress *netv1.Ingress) {
+				defaultAnnotations := map[string]string{
+					"nginx.ingress.kubernetes.io/proxy-body-size": "0",
+				}
+				require.Equal(t, defaultAnnotations, ingress.Annotations)
+				pathTypePrefix := netv1.PathTypePrefix
+				expectedRules := []netv1.IngressRule{
+					{
+						Host: "install.coder.com",
+						IngressRuleValue: netv1.IngressRuleValue{
```
Comment (johnstcn) [inline tests/ingress_test.go:37]: readability (non-blocking): Could you extract this struct to a variable?

File: tests/ingress_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	"github.com/jinzhu/copier"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIngress(t *testing.T) {
+	chart := LoadChart(t)
+
+	tests := []struct {
+		Name       string
+		ValuesFunc func(v *CoderValues)
```
Comment (johnstcn) [inline tests/ingress_test.go:18]: nit (non-blocking): Could you add a comment describing what this does?

**Review by johnstcn — APPROVED** [review]
I haven't tested this, but it looks fine to me. Some minor comments.

**Review by Emyrk — COMMENTED** [review]
Those tests look tedious to write!

File: tests/ingress_test.go (hunk 1 above)
Comment (jawnsy) [reply tests/ingress_test.go:37]: I'm confused, this whole struct is already a variable. Can you show me an example of what you mean?

Comment (johnstcn) [reply tests/ingress_test.go:37]: This is basically what I was thinking; all the values look to be the same:
```
var tcpCoderdRoot = netv1.IngressRuleValue{
    ...
}

...

expectedRules := []netv1.IngressRule{
    {
        Host: "xxx",
        IngressRuleValue: tcpCoderdRoot,
        ...
    },
...
```

Comment (jawnsy) [reply tests/ingress_test.go]: Extracted out some of the shared stuff, thanks!

File: tests/ingress_test.go (hunk 2 above)
Comment (jawnsy) [reply tests/ingress_test.go:41]: added!

**Comment by jawnsy** [comment]
>Those tests look tedious to write!
@Emyrk They are kinda tedious, yeah :frowning: But less tedious than always testing things manually, I think!

Do you have ideas for how to improve/simplify the API or test code? I'm open to suggestions!

I'm not in love with this code, but it's better than the alternative of manual tests only

**Comment by jawnsy** [comment]
Tested this in dev-4 and everything deploys fine, should be safe to merge

**Pull request merged**